src/plonkish/backend/halo2.rs

```diff
@@ -11,17 +11,27 @@ use halo2_proofs::{
 };
 
 use crate::{
-    ast::ToField,
+    field::Field as ChiquitoField,
     plonkish::ir::{
         assignments::Assignments,
         sc::{SuperAssignments, SuperCircuit},
         Circuit, Column as cColumn,
         ColumnType::{Advice as cAdvice, Fixed as cFixed, Halo2Advice, Halo2Fixed},
         PolyExpr,
     },
+    poly::ToField,
     util::UUID,
 };
 
+impl<T: Field + From<u64>> ChiquitoField for T {
+    const ZERO: Self = <Self as Field>::ZERO;
+    const ONE: Self = <Self as Field>::ONE;
+
+    fn mi(&self) -> Self {
+        self.invert().unwrap_or(Self::ZERO)
+    }
+}
+
```

Comment:
I don't see where we are using `T`.

Besides, looks like the only difference from ff::Field is that we return ZERO instead of error if the element inverted is zero.

What's the purpose of this again? To avoid using the is zero gadget?

Reply:
the purpose is that the core chiquito should not depend on halo2 traits as it should work with other backends.

This is a blanket implementation. It means that the trait is implemented for all Ts such that T: Field + From<u64>.

https://users.rust-lang.org/t/what-are-blanket-implementations/49904